service/controller/v18/cloudconfig/master_template.go

```diff
@@ -88,168 +94,235 @@ type MasterExtension struct {
 }
 
 func (e *MasterExtension) Files() ([]k8scloudconfig.FileAsset, error) {
+	var storageClass string
+	_, ok := e.encrypter.(*vault.Encrypter)
+	if ok {
+		storageClass = cloudconfig.InstanceStorageClassContent
+	} else {
+		storageClass = cloudconfig.InstanceStorageClassEncryptedContent
```

Comment:
Do you happen to know why we use encrypted storage only in case of vault?

Reply:
I think it is in other way. We don't use aws encryption in case of vault (because it handled by vault?)

Reply:
Oh yes you are right this is the other way around. But I still don't follow why this has to be changed.

Reply:
Maybe it doesn't work in China because there is no KMS. It would be nice to have that commented. @puja108 do you know?

Reply:
this should be conditional rather on china not on vault, vault is here a symptom of kms not being available in china, not very clean i would say, especially if at some point we would say we always use vault no matter if china or not